Osiris/extraHooks.cpp

```diff
@@ -0,0 +1,74 @@
+#include "extraHooks.h"
+#include "SDK/ModelInfo.h"
+
+typedef float quaternion[4];
+
+extraHooks extraHook;
+
+void __fastcall DoExtraBoneProcessing(Entity* player, uint32_t, StudioHdr* hdr, Vector* pos, quaternion* q, matrix3x4* matrix, void* bone_list, void* context)
+{
+	const auto entity = reinterpret_cast<Entity*> (player);;
+	auto state = entity->getAnimstate();
+	const auto val = reinterpret_cast<float*> (reinterpret_cast<uintptr_t> (state) + 292);
+	const auto backup = *val;
+	auto backup_onground = false;
+
+	if (state)
+	{
+		backup_onground = state->OnGround;
+		state->OnGround = false;
+
+		if (entity->velocity().length2D() < 0.1f)
+			*val = 0.f;
+	}
+
+	extraHook.player.vmt.getOriginal<void>(197, player, hdr, pos, q, matrix, bone_list, context);
```

Comment:
using MinHook and very inexperienced in coding. minhook getOriginal function cant identify which one is idx
in (getOriginal(197, player, hdr, pos, q, matrix, bone_list, context);)

Reply:
using vmt not minhook